lib/top/mdx_top.ml

```diff
@@ -39,7 +38,7 @@ let redirect ~f =
     read_up_to := pos;
     Buffer.add_channel buf ic len
   in
-  try_finally
+  Misc.try_finally
```

Comment:
I think we can be more future proof by porting uses of `Misc.try_finally` (in `compiler-libs` so technically unstable) to `Fun.protect` (in Stdlib, 4.08+). What do you think?

Reply:
:+1: `Fun.protect` in this case is a great 1:1 replacement, only needed to rename the keyword since it provides exactly the required semantics.